test/http.cc

```diff
@@ -0,0 +1,139 @@
+#include "stout/http.h"
+
+#include "event-loop-test.h"
+#include "gtest/gtest.h"
+#include "stout/eventual.h"
+#include "stout/interrupt.h"
+#include "stout/lambda.h"
+#include "stout/terminal.h"
+
+namespace eventuals = stout::eventuals;
+
+using eventuals::EventLoop;
+using eventuals::Interrupt;
+using eventuals::Terminate;
+
+using eventuals::http::Get;
+using eventuals::http::Post;
+using eventuals::http::PostFields;
+
+class HTTPTest : public EventLoopTest {};
+
+TEST_F(HTTPTest, GetFailTimeout) {
+  auto e = Get("http://example.com", std::chrono::milliseconds(1));
```

Comment:
If I understand correctly the plan is to replace all of these tests with the mock server but just wanted to point out that while unlikely the pattern you have in this tests and the ones below may lead to a flaky test and even if it's highly unlikely we still don't want to do it. But since I know the plan is to use the mock server I'm fine to start with these and then change to those in a subsequent PR.